Feature/helena lumi

DanHarJor opened the pull request:
I added some changes to the runner and parser of HELENA. 

Runner main change is that it can take a BETAN target as a parameter rather than being sampled

Parser so that it has the noKBM writer. 

I changed the beta itteration mechanisim, it is a linear prediction that corrects some mistakes of the previous one. It is also more clear that it is a linear prediction.

DanHarJor committed:
Tried to make a helena test, I keep getting errors, slurm fails with no reason given. I blame dask and am going to try and make a slurm array executor

DanHarJor committed:
I found some python errors, one was that teped was not included in sampler, amanda fixed it so that it doesn't have to be. The other was related to noKBM. It was difficult to get the python errors as I could not get the dask logs. Amanda showd me how you can alter the print so it goes to std out. I will impliment that soon. Still debugging.

DanHarJor committed:
Merge branch 'main' into feature/helena_lumi

DanHarJor committed:
I added the write parameters file no KBM

DanHarJor committed:
Merge branch 'feature/print_worker_python_errors_to_std_out' into feature/helena_lumi

DanHarJor committed:
I added some stuff to helena runner to allow beta_N to be set in the config file. I also added the ability to run it with the noKBM writer. I also made it so that for beta iteration style 1 the itteration results are written to a file, seems to be working, atleast some helena fort.20 files are being made.

DanHarJor committed:
update

DanHarJor committed:
I added a new test for the beta itteration

DanHarJor committed:
I added a print for the number of initial samples generated

DanHarJor committed:
update

DanHarJor committed:
small change to print to make more readible

DanHarJor committed:
update for deeplasma pipeline.

DanHarJor committed:
there was an issue where the dask workers had a different enchanted surrogates tot he one I was editing

DanHarJor committed:
There was an error in line 562 where a multiplyer was set to 1 which caused the beta_N to stay the same and then gave a divide by zero later. I changed it to 0.1.

DanHarJor committed:
I changed the current profile guess so that it doesn't use d_T_ped which is not always available

DanHarJor committed:
merge

DanHarJor committed:
update

DanHarJor committed:
update

DanHarJor committed:
update

DanHarJor committed:
fix o ftraceback print

DanHarJor committed:
changed the tolerance to 0.5 as I realised it is a fraction of the BETAN, I added a function for testing the HELENArunner without dask

DanHarJor committed:
After debugging with amanda in weekly meeting, we found a bug in the beta itteration where it was changing the parameter with a multiplier instead of setting to a new value for the linear fit. We also found an error in my choice of sample. We found that it works better if the 0.0 and 0.1 itteration of  at1 give a beta on either side of the target, therwise it is likeley to overshoot and cause helena to crash. Currently it will run but the beta is not going towards the target, more investigation needed

DanHarJor committed:
Merge branch 'feature/simple_lumi' into feature/helena_lumi

DanHarJor committed:
I updated the runner to have some new code for the linear predicion of at1 that gets betaN_target. The old version has some errors, (errors that are still present in the other option for beta iteration that modifies the fast ion pressure). It doesn't select which of the old points to do the linear prediction with, the linear prediction formula are also incorrect.

DanHarJor committed:
Small fix to point selection, syntax error

DanHarJor committed:
The beta itteration is now working for the at1 tuning. The other methods have been left and still have errors in the formula. I will get Amanda to check them.

DanHarJor commented:
There is an issue with the beta iterator formula. This was causing the next beta to be very large and causing Helena to crash.

I re-implimented the linear prediction based beta iterator and here is an example print out of its performance:
BETA ITERATION 0.0: target=2.555, current=0.596
BETA ITERATION 0.1: target=2.555, current=0.72
BETA ITERATION 0.2: target=2.555, current=2.472
BETA ITERATION 0: target=2.555, current=2.5749999999999997
BETA ITERATION 1: target=2.555, current=2.557
BETA ITERATION 2: target=2.555, current=2.554
BETA ITERATION 3: target=2.555, current=2.555
BETA ITERATION FINISHED.
Target betaN: 2.555
Final betaN: 2.555
Number of beta iterations: 4

DanHarJor committed:
Working very well with beta_iterations_afp=false, but HELENA crashed when I set it to true, even after the corrections to the linear prediction. I added the ability to set the input value from the parameters file for the initial linear prediction data points. I also changed constant beta variable to beta_N_target.

DanHarJor committed:
update

DanHarJor committed:
updates

DanHarJor committed:
silly error with self.input_value_2 not being there

awlishar commented:
Change input_value_1/2 to more descriptive :) also add method documentation

awlishar commented:
New dependencies need to be added to requirements file if they are not there

awlishar commented:
@DanHarJor have you tested the changes in the executor on Mahti for example or for the other runners/samplers?

DanHarJor committed:
I am doing this merge because my pull request has been there for a while
and I wanted to check it is still compatible with main
Merge branch 'main' into feature/helena_lumi

DanHarJor committed:
added some doccumentation

DanHarJor committed:
merge with main after target imports

DanHarJor committed:
ingoring log files *.log

DanHarJor committed:
I changed it so that it will make all the parent directories of the run dir if they don't exist

DanHarJor committed:
I added src to python path in the run.py dir. This way it does not need to be handled before running it

DanHarJor committed:
helena runner not working when using a sampler in normal enchanted surrogated workflow. Needs more debugging

DanHarJor committed:
tmp folders should not be committed or tracked

DanHarJor commented:
There seems to be some issues when using a sampler and a normal enchanted surrogates workflow. I will do some further debugging.

DanHarJor committed:
I found a bug that made only run run dir get created

DanHarJor committed:
I deleted the gene_parser_tests since it was causing pylint errors and belongs on another branch

DanHarJor committed:
I added some actual tests to check if betaN matched beta_n_target within the defined tolerance.

DanHarJor committed:
I added in the parts I was trying to move to profiles since the profiles are not perfected yet. I changed the python path add in run.py

DanHarJor committed:
small error with filename

DanHarJor committed:
Merge branch 'main' into feature/helena_lumi

DanHarJor commented:
I added a test where I take 16 sobel sequence samples of the pedestal and do betaN itterations on all of them.

I am happy with test because it is all we need for DEEPlasma purposes. Do you want to try it on Mahti or other tests?

I am happy to merge with main now.

awlishar commented on runLUMI.py:
```diff
@@ -0,0 +1,25 @@
+print('debug',profile['env']['ENCHANTED_SRC_PATH'])
+sys.path.append(profile['env']['ENCHANTED_SRC_PATH']) 
+
+import run
```
Imports not in the middle of script

awlishar commented on src/runners/HELENArunner.py:
```diff
@@ -517,69 +564,151 @@ def run_helena_with_beta_at1_iteration(self, run_dir, beta_target):
+        
+        
         subprocess.call([self.executable_path])
+        import shutil
```
Don't put imports in the middle of functions

awlishar commented on tests/LUMI_tests/helena_old.py:
Is it necessary to have an "_old" test file? The old version should be retrievable from the git history

awlishar commented on src/runners/HELENArunner.py:
Let's both (@DanHarJor and me) test this version before merging

awlishar reviewed (commented):
Needs testing and some code fixes
